os/pm/pm_metrics.c

```diff
@@ -84,14 +85,12 @@ static void pm_print_metrics(double total_time, int n_domains)
 	pmdbg("\n");
 	pmdbg(" BOARD STATE | PM STATE |          TIME          \n");
 	pmdbg("-------------|----------|------------------------\n");
-	pmdbg(" %11s | %8s | %10dms (%6.2f%%) \n", "WAKEUP", "NORMAL", TICK2MSEC(g_pm_metrics->state_metrics.state_accum_ticks[0]),
-		  ((double)g_pm_metrics->state_metrics.state_accum_ticks[0]) * 100.0 / total_time);
-	pmdbg(" %11s | %8s | %10dms (%6.2f%%) \n", "", "IDLE", TICK2MSEC(g_pm_metrics->state_metrics.state_accum_ticks[1]),
-		  ((double)g_pm_metrics->state_metrics.state_accum_ticks[1]) * 100.0 / total_time);
-	pmdbg(" %11s | %8s | %10dms (%6.2f%%) \n", "", "STANDBY", TICK2MSEC(g_pm_metrics->state_metrics.state_accum_ticks[2]),
-		  ((double)g_pm_metrics->state_metrics.state_accum_ticks[2]) * 100.0 / total_time);
+	for (pm_state = PM_NORMAL; pm_state < PM_SLEEP; pm_state++) {
+		pmdbg(" %11s | %8s | %10dms (%6.2f%%) \n", ((pm_state == PM_NORMAL) ? "WAKEUP" : ""), pm_state_name[pm_state], TICK2MSEC(g_pm_metrics->state_metrics.state_accum_ticks[0]),
+			((double)g_pm_metrics->state_metrics.state_accum_ticks[pm_state]) * 100.0 / total_time);
+	}
 	pmdbg("-------------|----------|------------------------\n");
-	pmdbg(" %11s | %8s | %10dms (%6.2f%%) \n", "SLEEP", "SLEEP", TICK2MSEC(g_pm_metrics->board_sleep_ticks),
+	pmdbg(" %11s | %8s | %10dms (%6.2f%%) \n", "SLEEP", pm_state_name[PM_SLEEP], TICK2MSEC(g_pm_metrics->board_sleep_ticks),
```

Comment:
"SLEEP", pm_state_name[PM_SLEEP],

It looks both are the same string. If yes, why do you use different way?

Reply:
The left side represents board state while right side represents pm state. For board `WAKEUP` state, we have three pm states i.e. `NORMAL, IDLE, STANDBY`. But for board `SLEEP` state, we only have one pm state i.e. `SLEEP`.  We keep them different because they represent different meaning.

Reply:
You mean, because the left is the board state, it's not good to use the pm_state_name, even they are the same strings. Do I understand well?

Reply:
yes